servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -376,6 +377,12 @@ private Throwable enrichError(final Throwable t) {
         } else {
             if ((closeReason = this.closeReason) != null) {
                 throwable = wrapWithCloseReason(closeReason, t);
+                if (closeReason == CHANNEL_CLOSED_INBOUND && t instanceof StacklessClosedChannelException &&
+                        !NativeTransportUtils.isEpollAvailable() && !NativeTransportUtils.isKQueueAvailable()) {
+                    throwable = new StacklessClosedChannelException("Observed closed channel after an attempt to " +
```

Comment:
@Scottmitch wdyt about this approach?

Reply:
I like the idea of providing more context in exceptions. this specific implementation seems somewhat indirect and potentially fragile (e.g. exceptions types are pinned, is the the only place the exception needs to be handled, can the exception be wrapped in anything else before this point), but I haven't thought about an alternative.